webapp/tests/karma/ts/services/telemetry.service.spec.ts

```diff
@@ -227,23 +201,121 @@ describe('TelemetryService', () => {
         },
         deviceInfo: {}
       });
+
       expect(dbInstance.query.callCount).to.equal(1);
       expect(dbInstance.query.args[0][0]).to.equal('medic-client/doc_by_type');
       expect(dbInstance.query.args[0][1]).to.deep.equal({ key: ['form'], include_docs: true });
       expect(pouchDb.destroy.callCount).to.equal(1);
       expect(pouchDb.close.callCount).to.equal(0);
+
+      expect(consoleErrorSpy.callCount).to.equal(0);  // no errors
     });
 
+    it('should not aggregate when recording the day the db was created and next day it should aggregate', async () => {
+      setupDbMocks();
+      storageGetItemStub.withArgs('medic-greg-telemetry-date').returns(sameDay());
+
+      await service.record('test', 10);
+
+      expect(pouchDb.post.callCount).to.equal(1);
+      expect(pouchDb.post.args[0][0]).to.deep.include({ key: 'test', value: 10 });
+      expect(dbInstance.put.callCount).to.equal(0);   // NO telemetry has been recorded yet
+
+      clock = sinon.useFakeTimers(moment(NOW).add(1, 'minutes').valueOf()); // 1 min later ...
+      await service.record('test', 5);
+
+      expect(pouchDb.post.callCount).to.equal(2); // second call
+      expect(pouchDb.post.args[1][0]).to.deep.include({ key: 'test', value: 5 });
+      expect(dbInstance.put.callCount).to.equal(0);   // still NO telemetry has been recorded (same day)
+
+      clock = sinon.useFakeTimers(moment(NOW).add(1, 'days').valueOf()); // 1 day later ...
+      await service.record('test', 2);
+
+      expect(pouchDb.post.callCount).to.equal(3); // third call
+      expect(pouchDb.post.args[2][0]).to.deep.include({ key: 'test', value: 2 });
+      expect(dbInstance.put.callCount).to.equal(1);   // Now telemetry has been recorded
+
+      const aggregatedDoc = dbInstance.put.args[0][0];
+      expect(aggregatedDoc._id).to.match(/telemetry-2018-11-10-greg/);  // Now is 2018-11-11 but aggregation
+      expect(pouchDb.destroy.callCount).to.equal(1);                    // is from from previous day
+
+      expect(consoleErrorSpy.callCount).to.equal(0);  // no errors
+    });
+
+    it('should aggregate from days with records skipping days without records', async () => {
```

Comment:
I don't think it is a good idea to send aggregation from days that the app was not used, that can be inferred by the consumers of the telemetry when they query the data. It will increase the size of data sent and stored, and although it shouldn't be a big deal, it does not add any value IMHO.

We do send monthly aggregation that also can lead to gaps now, although is less likely a user not using the app the whole month, but if that happens, the result is the same, only telemetry of the last month when the app was used is sent in the current monthly implementation, and I just wanted to add test coverage to this scenario, to keep that in mind when changing the code.

If we want to move forward with "null" aggregation reports, I would need to change how we aggregate data, because the data needs to be grouped first by date, instead of making one report calling the built-in "_stats" reducers.... though because there is nothing to aggregate, maybe we can just duplicate the report gotten from the last day, dropping all the real data and only kept the "metadata" section, but again I don't think that this is a good idea.